motionplan/tpspace/ptgIK.go

```diff
@@ -114,38 +131,48 @@ func (ptg *ptgIK) MaxDistance() float64 {
 }
 
 func (ptg *ptgIK) Trajectory(alpha, dist float64) ([]*TrajNode, error) {
+	traj := []*TrajNode{}
 	ptg.mu.RLock()
 	precomp := ptg.trajCache[alpha]
 	ptg.mu.RUnlock()
-	if precomp != nil {
-		if precomp[len(precomp)-1].Dist >= dist {
-			// Caching here provides a ~33% speedup to a solve call
-			subTraj := []*TrajNode{}
-			for _, wp := range precomp {
-				if wp.Dist < dist {
-					subTraj = append(subTraj, wp)
-				} else {
-					break
+	if precomp != nil && precomp[len(precomp)-1].Dist >= dist {
+		// Caching here provides a ~33% speedup to a solve call
```

Comment:
It looks like this shouldn't be applied to this line but to where you define the traj slice